keps/sig-auth/3766-referencegrant/README.md

```diff
@@ -0,0 +1,920 @@
+
+Before this graduates to beta, we will provide a reference implementation with a
+sample CRD that will be used to provide integration tests.
+
+##### e2e tests
+
+We will strongly encourage every API that uses ReferenceGrant to define
+conformance tests for their use of ReferenceGrant.
+
+### Graduation Criteria
+
+#### Beta
+
+[ ] Reference implementation with integration tests.
+[ ] Almost all of the fields and behavior have conformance test coverage in at least one project (for example Gateway API).
+
+#### GA
+
+[ ] Conformance tests that exercise all ReferenceGrant API calls (not the actual implementation of the API).
+[ ] Multiple implementations of this API passing all relevant conformance tests.
+
+### Upgrade / Downgrade Strategy
+
+N/A
+
+### Version Skew Strategy
+
+Version skew is a bit different here. Although we will provide a shared library
+for implementing this API, this will only be used by third-party controllers,
+not built-in components.
+
+There will be some implementations that support both the API defined by Gateway
+API and this API. Since these resources are entirely additive and can be
+duplicative, we can copy Gateway API resources to the new API group and delete
+the old Gateway API resources as part of a seamless migration. We expect that
+many implementations will provide this recommendation to users, and we may even
+provide tooling to simplify this process.
+
+<<[UNRESOLVED open questions that don't clearly fit elsewhere ]>>
+## Open Questions
+
+This KEP was merged in a provisional state with a number of open questions
+remaining. This section highlights the questions we have not resolved yet.
+
+### Do We Need Verbs?
+
+Previous Discussion: https://github.com/kubernetes/enhancements/pull/3767#discussion_r1084509958
+
+We could add a `verbs` field to enable users to specify the kind of referential
+access they want to grant. For example, we could define "read", "route", and
+"backup" as well-known verbs to start. We could also allow implementations to
+support additional domain-prefixed verbs.
+
+This would enable more precise grants, and potentially more open ended fields
+elsewhere in the resource, see the next item for more.
+
+### Do We Need Any or Wildcard Selectors?
+
+Previous Discussion: https://github.com/kubernetes/enhancements/pull/3767#discussion_r1086020464
+
+We already allow "Name" to be optional in `To`, effectively resulting in
+wildcard behavior. Should we expand that to allow any of the following?
+
+1. References to any group or resource
+1. References from any group or resource
+1. References from any namespace
+
+Historically, there has been pushback to allowing any group or resource
+because it could enable unknown or unintended kinds of access. If we
+introduced the concepts of "verbs" described above, this would become
+a moot point.
+
+### Do We Need Label Selectors?
+
+Previous Discussion: https://github.com/kubernetes/enhancements/pull/3767#discussion_r1084492070
```

Comment:
https://github.com/kubernetes/enhancements/pull/3767#discussion_r1084674648 was another thread about selectors you could link to as well

Reply:
Updated, thanks!